gap/attr.gi

```diff
@@ -1582,3 +1582,119 @@ function(gr)
   od;
   return fail;
 end);
+
```

Comment:
Can you give some information about where this algorithm comes from?

Reply:
Yes, indeed, I second @wilfwilson's comment here. You should give a link to wherever you read about this algorithm here, so that anyone looking at this can also read about it. You should probably also add a citation to the documentation, stating where this is taken from.